packages/hardhat-core/src/internal/cli/cli.ts

```diff
@@ -60,6 +64,38 @@ To learn more about which versions of Node.js are supported go to https://hardha
   }
 }
 
+async function suggestInstallingHardhatVscode() {
+  const alreadyPrompted = hasPromptedForHHVSCode();
+  if (alreadyPrompted) {
+    return;
+  }
+
+  const isInstalled = isHardhatVSCodeInstalled();
+
```

Comment:
I think we should add something like this:

```js
  if (isInstalled === InstallationState.EXTENSION_INSTALLED) {
    writePromptedForHHVSCode();
    return;
  }
```

Otherwise the `test` task will be a bit slower for all Hardhat VSCode users. (I don't know if the difference would be noticeable, but it feels like the right thing to do anyway.)

Reply:
Update: on my machine, if you already have the extension installed, it adds at least 300-700 ms at the end of the test task.

Reply:
Done. Now I always store it.